src/main/java/technology/dice/dicewhere/provider/dbip/parsing/DbIpIpToCityCSVLineParser.java

```diff
@@ -0,0 +1,56 @@
+package technology.dice.dicewhere.provider.dbip.parsing;
+
+import com.google.common.base.Splitter;
+import com.google.common.net.InetAddresses;
+import technology.dice.dicewhere.api.api.IP;
+import technology.dice.dicewhere.api.api.IpInformation;
+import technology.dice.dicewhere.api.exceptions.LineParsingException;
+import technology.dice.dicewhere.parsing.LineParser;
+import technology.dice.dicewhere.parsing.ParsedLine;
+import technology.dice.dicewhere.reading.RawLine;
+import technology.dice.dicewhere.utils.StringUtils;
+
+import java.net.InetAddress;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class DbIpIpToCityCSVLineParser implements LineParser {
+  private static final Splitter splitter = Splitter.on(',');
+
+  @Override
+  public ParsedLine parse(RawLine line, boolean retainOriginalLine) throws LineParsingException {
+    try {
+      Iterable<String> fieldsIterable = splitter.split(line.getLine());
+      Iterator<String> fieldsIterator = fieldsIterable.iterator();
+      String rangeStartString = StringUtils.removeQuotes(fieldsIterator.next());
+      String rangeEndString = StringUtils.removeQuotes(fieldsIterator.next());
+      fieldsIterator.next(); // skip continent
+      String countryCode = fieldsIterator.next();
+      String leastSpecificDivision = fieldsIterator.hasNext() ? fieldsIterator.next() : null;
+      String city = fieldsIterator.hasNext() ? fieldsIterator.next() : null;
```

Comment:
null or empty string?

Reply:
I am going with your previous advise of using null.
Have you changed your mind?

Reply:
Will be much better once it's using the [builder](#8) instead